lib/Conversion/ExportChiselInterface/ExportChiselInterface.cpp

```diff
@@ -32,21 +32,22 @@ static const unsigned int indentIncrement = 2;
 
 /// Emits type construction expression for the port type, recursing into
 /// aggregate types as needed.
-static LogicalResult emitPortType(Location location, FIRRTLBaseType type,
+static LogicalResult emitPortType(Location location, FIRRTLType type,
                                   Direction direction, llvm::raw_ostream &os,
-                                  unsigned int indent,
+                                  bool &containsProbeType, unsigned int indent,
                                   bool hasEmittedDirection = false) {
   auto emitTypeWithArguments =
       [&](StringRef name,
           // A lambda of type (bool hasEmittedDirection) -> LogicalResult.
           auto emitArguments,
           // Indicates whether parentheses around type arguments should be used.
           bool emitParentheses = true) -> LogicalResult {
-    // Include the direction if the type is not composed of flips and analog
-    // signals and we haven't already emitted the direction before recursing to
-    // this field.
+    // Include the direction if the type is a ref type or is not composed of
+    // flips and analog signals and we haven't already emitted the direction
+    // before recursing to this field.
     bool emitDirection =
-        type.isPassive() && !type.containsAnalog() && !hasEmittedDirection;
+        type.isa<RefType>() ||
+        (type.isPassive() && !type.containsAnalog() && !hasEmittedDirection);
```

Comment:
Hyper-nit: Should `!hasEmittedDirection`  go first as it's local and trivial to check (and when going through recursive types will often be false?).

I haven't thought much about it, but what's the reasoning here? Why only emit direction for passive things?
Analog kinda is weird re:being legal to be `input` vs `output` at all.  Is output analog not reachable from Chisel?

Reply:
Good point about `hasEmittedDirection`.

Chisel direction functions override any internal directions. So doing `Output(new Bundle {...})` erases all direction information inside the bundle. That's why directions are only placed on the innermost passive members. And they otherwise don't matter for `Analog`.

Reply:
I'll add a comment explaining that

Reply:
Awesome, thanks for the explanation re:Chisel.  TIL! :+1: